### ISSUE TITLE
[Feature Request]: non-blocking implementation of `nativeImage.createThumbnailFromPath()` for Windows

### ISSUE THREAD

**Issue by RaisinTen** [electron/electron#31024]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Problem Description

`nativeImage.createThumbnailFromPath()` is implemented in a blocking way for Windows https://github.com/electron/electron/blob/efa70131e2b88031470f8cc4ec5a8e166c17615b/shell/common/api/electron_api_native_image_win.cc#L25

### Proposed Solution

This and all other asynchronous functions should be implemented in a non-blocking way

### Alternatives Considered

None

**Comment by codebytere** [comment]
@RaisinTen can you elaborate on this? 

> This and all other asynchronous functions should be implemented in a non-blocking way

Do you mean windows functions? On macOS for example this is non-blocking.

**Comment by RaisinTen** [comment]
Yes @codebytere that is right, for the `nativeImage.createThumbnailFromPath()` function, I was only talking about the Windows implementation. However, if we find out more such async functions that block the current thread, we should consider dispatching the task to some other thread.

**Comment by seb-mtl** [comment]
Can you elaborate on this? I am planning on using this function. Is this blocking the main thread despite being async?

**Comment by RaisinTen** [comment]
@seb-mtl yes, that's right! When you call this function, it would run the sequence of function calls inside `NativeImage::CreateThumbnailFromPath()` synchronously, which blocks the current thread. It's only the resolution/rejection of the promise that takes place as a microtask. We need to dispatch these calls to a different thread (in Chromium's thread pool) to make this non-blocking. For reference, I had sent https://github.com/electron/electron/pull/31023 earlier to create a non-blocking version of a pre-existing function! ;)

**Comment by seb-mtl** [comment]
@RaisinTen Thanks a lot! You just saved me from hours of debugging.😉
